### PR TITLE
Run integration tests against Maistra/istio

### PR DESCRIPTION
**Pull request by dgn**

### CODE REVIEW

**Comment by dgn** [comment]
Integration tests for both `maistra-1.1`[1] and `maistra-1.2`[2] branches now pass on prow!

[1]: https://prow.maistra.io/view/gcs/maistra-prow-testing/pr-logs/pull/Maistra_istio/123/maistra-istio-1.1-integration/1273260181202931713
[2]: https://prow.maistra.io/view/gcs/maistra-prow-testing/pr-logs/pull/Maistra_istio/115/maistra-istio-1.2-integration/1273238598962909184

File: docker/maistra-builder_1.2.Dockerfile [hunk 1]
```diff
@@ -1,10 +1,11 @@
 FROM fedora:31
 
 # Versions
-
+ENV GO_VERSION="1.14.4"
```
Comment (dgn) [inline docker/maistra-builder_1.2.Dockerfile:4]: Istio 1.6 integration tests require golang 1.14 (actually it's only one test that does, but well...)

File: docker/scripts/entrypoint.sh [hunk 2]
```diff
@@ -1,5 +1,10 @@
 #!/bin/bash
 
+DOCKER_DAEMON_ARGS=${DOCKER_DAEMON_ARGS:-"-l warn"}
+
+# Inject CloudFlare DNS to make sure we don't have DNS loops
+echo "nameserver 1.1.1.1" > /etc/resolv.conf
```
Comment (dgn) [inline docker/scripts/entrypoint.sh:6]: without this part, the kind cluster wasn't able to pull images

File: docker/maistra-builder_1.2.Dockerfile (hunk 1 above)
Comment (jwendell) [reply docker/maistra-builder_1.2.Dockerfile:4]: What about upgrading the base image to Fedora 32? It has go 1.14.

File: docker/maistra-builder_1.2.Dockerfile [hunk 3]
```diff
@@ -39,11 +44,22 @@ RUN curl https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz | tar -xz l
...
+    chmod +x /usr/local/bin/kind
+
+# gcloud
+# some upstream scripts rely on gcloud commands to retrieve credentials for
+# e.g. pushing to GCS buckets. We don't require this at this point, so we're
+# only creating a stub script to make sure scripts don't fail because gcloud
+# is missing.
+RUN echo 'echo "gcloud is not installed. Skipping"' > /usr/local/bin/gcloud && \
+    chmod +x /usr/local/bin/gcloud
+
```
Comment (jwendell) [inline docker/maistra-builder_1.2.Dockerfile:62]: Really? Those should be optional, like here: https://github.com/istio/istio/blob/master/prow/lib.sh#L32
If they aren't please, fix that upstream :)

Comment (dgn) [reply docker/maistra-builder_1.2.Dockerfile:62]: They are now, but were not in 1.4: https://github.com/istio/istio/blob/cf35a864ff65ae3a6d0278fa0c5bfead0eaafffe/prow/lib.sh#L38

I guess we could fix it in our fork, too. But I don't think changing that upstream now makes a lot of sense

File: docker/maistra-builder_1.2.Dockerfile (hunk 1 above)
Comment (dgn) [reply docker/maistra-builder_1.2.Dockerfile:4]: Good point, but if possible I'd like to do that in a follow-up. I finally got this working and don't want to get it stuck on upgrade issues. As soon as it's rolled out I can look at upgrading to f32

Comment (jwendell) [reply docker/maistra-builder_1.2.Dockerfile:4]: Fair enough

File: docker/maistra-builder_1.2.Dockerfile (hunk 3 above)
Comment (jwendell) [reply docker/maistra-builder_1.2.Dockerfile:62]: We should not be doing that, that's the good thing about using branches, one can evolve independently.

**Pull request merged**